typescript/web/src/components/workspace-name-input/workspace-name-message.tsx

```diff
@@ -17,10 +18,10 @@ const useUrlMessage = (
   isEditing: boolean | undefined
 ): string | undefined => {
   const { slug } = useWorkspaceNameInput();
-  const hideUrl =
-    !isEmpty(error) || isEmpty(slug) || isEmpty(globalThis.location);
+  const { origin } = useMockableLocation();
```

Comment:
While would I agreed with this before working on this issue, the thing is that:
* Storybook has no way to override the hostname used by its stories via Location (like `siteUrl` for Jest tests).
* We can't easily override Location only for an inner React component and its callbacks.
* `globalThis.location` and `window.location` are used at other places in the code and possibly by some dependencies.
* It's returned values don't trigger re-renders when they're changed as you'd expect with a regular `useLocation` hook (e.g. [react-router version](https://v5.reactrouter.com/web/api/Hooks/uselocation)).

As such, this hooks and its provider are really only useful when you need to use an instance Location which can me mocked. You don't really need it in other cases.

Also, the component still don't have the responsibility to know whether it is mocked or not (that's why I used `mockable` instead of `mocked` in the name). It's under the responsibility of the component which has declared the correct provider (the step wrapper or the main app entrypoint).